web/src/main/ui/app/components/flows-new/edit-flow/ui/new-step-dialog-ui.component.html

```diff
@@ -3,31 +3,31 @@ <h1 mat-dialog-title>{{title}}</h1>
 
   <mat-form-field class="ng-invalid">
     <mat-select id="step-type" class="type-select" placeholder="Step Type" (selectionChange)="stepTypeChange()"
-                formControlName="type" required>
+                formControlName="stepDefinitionType" required>
       <mat-option class="type-option" *ngFor="let option of stepOptions" [value]="option">{{option}}</mat-option>
     </mat-select>
   </mat-form-field>
-  <div *ngIf="newStepForm.value.type">
+  <div *ngIf="newStepForm.value.stepDefinitionType">
   <mat-form-field class="">
     <input id="step-name" matInput formControlName="name" placeholder="Name" required>
   </mat-form-field>
   <mat-form-field class="">
     <input id="step-description" matInput formControlName="description" placeholder="Description">
     <mat-hint>Optional</mat-hint>
   </mat-form-field>
-  <div *ngIf="newStepForm.value.type != 'ingest'">
+  <div *ngIf="newStepForm.value.stepDefinitionType !== 'INGESTION'">
     <label>Source Type: </label>
     <mat-radio-group id="step-source-type" [(ngModel)]="selectedSource" [ngModelOptions]="{standalone: true}">
       <mat-radio-button id="step-source-type-collection-radio" value="collection">Collection</mat-radio-button>
       <mat-radio-button id="step-source-type-query-radio" value="query">Query</mat-radio-button>
     </mat-radio-group>
   </div>
-  <mat-form-field *ngIf="selectedSource === 'collection' && newStepForm.value.type != 'ingest'">
-    <mat-select id="step-source-collection" class="type-select" placeholder="Source Collection">
+  <mat-form-field *ngIf="selectedSource === 'collection' && newStepForm.value.stepDefinitionType !== 'INGESTION'">
+    <mat-select id="step-source-collection" class="type-select" placeholder="Source Collection" formControlName="sourceQuery">
```

Comment:
This will still be stored as sourceCollection in the UI model, but we will need to add a property sourceQuery to that model that is a string based on the collection selection made here.  See more here: https://project.marklogic.com/jira/browse/DHFPROD-2177